lib/types/Internal.types.ts

```diff
@@ -272,3 +90,83 @@ export class InternalPresentationDefinitionV2 implements PresentationDefinitionV
     return PEVersion.v2;
   }
 }
+
+export interface WrappedVerifiableCredential {
+  /**
+   * Original VC that we've received
+   */
+  original: string | JwtWrappedVerifiableCredential | IVerifiableCredential;
+  /**
+   * In case of JWT credential it will be the decoded version. In other cases it will be the same as original one
+   */
+  decoded: JwtWrappedVerifiableCredential | IVerifiableCredential;
+  /**
+   * Type of this credential. Supported types are json-ld and jwt
+   */
+  type: VerifiableDataExchangeType;
+  /**
+   * created based on https://www.w3.org/TR/vc-data-model/#jwt-decoding
+   */
+  internalCredential: InternalCredential;
+}
+
+export interface WrappedVerifiablePresentation {
+  original: string | JwtWrappedVerifiablePresentation | IVerifiablePresentation;
+  decoded: JwtWrappedVerifiablePresentation | IVerifiablePresentation;
+  type: VerifiableDataExchangeType;
+  internalPresentation: InternalPresentation;
+  vcs: WrappedVerifiableCredential[];
+}
+
+export enum VerifiableDataExchangeType {
+  JSONLD,
+  JWT_ENCODED,
+  JWT_DECODED,
+}
+
+export interface InternalPresentation {
+  '@context': ICredentialContextType | ICredentialContextType[];
+  type: string[];
+  verifiableCredential: WrappedVerifiableCredential[];
+  presentation_submission?: PresentationSubmission;
+  holder?: string;
+}
+
+export interface InternalCredential {
```

Comment:
Is there a specific reason why this is called InternalCredential instead of ICredential (and moved over to the other SSI-types file)?
Looking at it quicly this seems to fit the interface for a normal Credential. Nothing internal about it that I can see

Reply:
yeah, it's the Json-LD credential and also the converted jwt-credential from https://www.w3.org/TR/vc-data-model/#jwt-decoding

Reply:
Okay but is the interface adding or lacking something which a normal Credential would not have?

Reply:
no, this is our normal credential. and with `& IHasProof` we will have a VerifiableCredential